e2e/nomostest/gitproviders/repository.go

```diff
@@ -554,7 +549,11 @@ func (g *Repository) CommitAndPushBranch(msg, branch string) error {
 		return err
 	}
 	g.Logger.Infof("[repo %s] pushing commit: %s", path.Base(g.Root), commit)
-	return g.Push(branch, "-f")
+	if err := g.Push(branch, "-f"); err != nil {
+		return err
+	}
+	g.isEmpty = false
```

Comment:
`isEmpty = false` -> `initialized = true`

The description on the field is technically incorrect too. It's not whether there are any commits (because there might be some pre-exiting, when attaching to an existing remote repo). It's actually just whether `Init` has been called yet, which force-pushes on top of any optionally existing commits to "reset" the contents of the repo to the default contents.